### PR TITLE
tests/bsim: Add a script to generate coverage reports

### PR DESCRIPTION
**Pull request by aescolar**
For users convenience, as it is a typical usecase to want to see the actual coverage of a set of tests.

Related to #56421

### CODE REVIEW

File: tests/bsim/_generate_coverage_report.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  echo "\
+_generate_coverage_report.sh [-help]
+  Generate an html coverage report for BabbleSim tests
+
+  You can call it from the Zephyr top level as:
+    tests/bsim/_generate_coverage_report.sh
+
+  Coverage files will be searched for in the folder pointed by
+  the variable WORK_DIR which by default is
+  ${ZEPHYR_BASE}/bsim_out
```
Comment (alxelax) [inline tests/bsim/_generate_coverage_report.sh:17]: Will the script generate coverage only for mesh if my `bsim_out` has only mesh compilation result? Or I need specify mesh path in any case?

File: tests/bsim/README.md [hunk 2]
```diff
@@ -38,6 +38,13 @@ tests/bsim/net/sockets/echo_test/tests_scripts/echo_test_802154.sh
 ```
 
 
+Generating coverage reports
+---------------------------
+
+You can use tests/bsim/_generate_coverage_report.sh to generate a coverage report
+of your BabbleSim tests. Check its help for more info.
```
Comment (alxelax) [inline tests/bsim/README.md:45]:
```suggestion
of BabbleSim tests. Check its help for more info.
```
what if I want to know a coverage not only my tests :)

File: tests/bsim/_generate_coverage_report.sh (hunk 1 above)
Comment (aescolar) [reply tests/bsim/_generate_coverage_report.sh:17]: >Will the script generate coverage only for mesh if my bsim_out has only mesh compilation result

Yes. It just collects the coverage data that is there. So if you only run mesh tests you only get mesh tests coverage data generated, and only that processed into the report.

**Review by jori-nordic — DISMISSED** [review]
works on my machine (TM)

File: tests/bsim/README.md (hunk 2 above)
Comment (aescolar) [reply tests/bsim/README.md]: rephrased :)

**Comment by aescolar** [comment]
@jori-nordic @alxelax thanks for the reviews. I rephrased what @alxelax did not like, and renamed to script to not have an underscore and instead added it to the run_parallel filter.
Here the diff compared to what you reviewed:
https://github.com/zephyrproject-rtos/zephyr/compare/9d2c26f9c4fbe887cf553952ca4c03b33b6e40ad..46cff745f06df23d2ee0e33132386e1836bb8b54

**Review by alxelax — APPROVED** [review]

**Review by jori-nordic — APPROVED** [review]

**Pull request merged**